projects/geometry/102_color_cube_vulkan/102_color_cube_vulkan.cpp

```diff
@@ -0,0 +1,503 @@
+#include "window.h"
+
+#include "vk_renderer.h"
+#include "tri_mesh.h"
+
+#include <glm/glm.hpp>
+#include <glm/matrix.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/gtx/transform.hpp>
+using namespace glm;
+
+#define CHECK_CALL(FN)                               \
+    {                                                \
+        VkResult vkres = FN;                         \
+        if (vkres != VK_SUCCESS) {                   \
+            std::stringstream ss;                    \
+            ss << "\n";                              \
+            ss << "*** FUNCTION CALL FAILED *** \n"; \
+            ss << "FUNCTION: " << #FN << "\n";       \
+            ss << "\n";                              \
+            GREX_LOG_ERROR(ss.str().c_str());        \
+            assert(false);                           \
+        }                                            \
+    }
+
+// =============================================================================
+// Shader code
+// =============================================================================
+const char* gShaderVS= R"(
+#version 460
+
+layout( push_constant ) uniform CameraProperties 
+{
+	mat4 MVP;
+} cam;
+
+in vec3 PositionOS;
+in vec3 Color;
+
+out vec3 vertexColor;	// Specify a color output to the fragment shader
+
+void main()
+{
+	gl_Position = cam.MVP * vec4(PositionOS, 1);
+	vertexColor = Color;
+}
+)";
+
+const char* gShaderFS= R"(
+#version 460
+
+in vec3 vertexColor;	// The input variable from the vertex shader (of the same name)
+
+out vec4 FragColor;
+
+void main()
+{
+	FragColor = vec4(vertexColor, 1.0f);
+}
+)";
+
+// =============================================================================
+// Globals
+// =============================================================================
+static uint32_t gWindowWidth        = 1280;
+static uint32_t gWindowHeight       = 720;
+static bool     gEnableDebug        = true;
+static bool     gEnableRayTracing   = false;
+static uint32_t gUniformmBufferSize = 256;
+
+void CreatePipelineLayout(VulkanRenderer* pRenderer, VkPipelineLayout* pLayout);
+void CreateRenderPass(VulkanRenderer* pRenderer, VkRenderPass* pRenderPass);
```

Comment:
If you use VK_KHR_dynamic_rendering - render passes and framebuffers can be skipped all together.

Reply:
Changes added to the pull request!